### PR TITLE
Use sidekiq for background jobs

### PR DESCRIPTION
**Pull request by iHiD**
I'll do Production next.

You get a dashboard at http://localhost:3020/sidekiq/ (or whatever your local host is)

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]
LGTM. Shall I do the development-environment part?

**Comment by iHiD** [comment]
Yes please!

File: config/initializers/sidekiq.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Sidekiq.configure_server do |config|
+  config.redis = { url: Exercism.config.sidekiq_redis_url }
```
Comment (SleeplessByte) [inline config/initializers/sidekiq.rb:2]: I don't know if this `raises` if this value isn't set, but it probably should (or the server is going to hang a bit later, which is a weird error)

File: config/routes.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
+require 'sidekiq/web'
+
 Rails.application.routes.draw do
   mount ActionCable.server => '/cable'
+  authenticate :user, ->(user) { user.admin? } do
+    mount Sidekiq::Web => '/sidekiq'
```
Comment (SleeplessByte) [inline config/routes.rb:6]:
```suggestion
    mount Sidekiq::Web => '/jobs'
```

Personal preference. Allows you to later change the UI/backing job system without changing the URL :D

**Review by SleeplessByte — APPROVED** [review]

File: config/initializers/sidekiq.rb (hunk 1 above)
Comment (iHiD) [reply config/initializers/sidekiq.rb:2]: Good shout.

File: config/routes.rb (hunk 2 above)
Comment (iHiD) [reply config/routes.rb:6]: I'm ok with this for now, as it lets me try multiple backends in parallel. But I might shift before launch. Thanks for the suggestion.

File: config/initializers/sidekiq.rb (hunk 1 above)
Comment (iHiD) [reply config/initializers/sidekiq.rb:2]: Done in Exercism Config v0.58.

**Pull request merged**